### ISSUE TITLE
Confirm: use of disabled attribute inside dataTable produces wrong behavior.

### ISSUE THREAD

**Issue by lagar84** [primefaces/primefaces#4125]
## 1) Environment
- PrimeFaces version: 6.2.10
- Does it work on the newest released PrimeFaces version? Version? No
- Does it work on the newest sources in GitHub? No
- Application server + version: jetty from primefaces-test (not important)
- Affected browsers: all (server side bug)

## 2) Expected behavior
When disabled attribute evaluates to true, no confirmDialog should appear. The correct row will be processed.

## 3) Actual behavior
On the first row that disabled evaluates to false, the remaining rows that disabled evaluates to true will be replaced by the previous row that disabled evaluated to false.

## 4) Steps to reproduce
Use this reproducer: https://github.com/lagar84/primefaces-test
http://localhost:8080/primefaces-test/confirmDialogTest.xhtml
In the reproducer:
   1) In the first row, click on "click me" link. disabled evaluates to true, and it will not show confirmDialog. Below dataTable will appear the message "You clicked: Mojarra". So far so good.
   2) On the second row, disabled evaluates to false, so the confirmDialog will appear. Click yes. Below dataTable will appear "You clicked: Myfaces". OK
   3) Click on the fifth row (Spring). It will show a confirmDialog, but it should not. Click yes. Below dataTable, will appear "You clicked: Hibernate" Note that the fourth row assume the fifth row place.

## 5) Sample XHTML
```xhtml
<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml"
      xmlns:ui="http://java.sun.com/jsf/facelets"
      xmlns:f="http://java.sun.com/jsf/core"
      xmlns:p="http://primefaces.org/ui"
      xmlns:h="http://java.sun.com/jsf/html">

    <h:head>
        <title>PrimeFaces Test</title>
    </h:head>
    <h:body>

		<h3>Problem:</h3>
		<p>After the first enabled confirmDialog (disabled=false), the remaing rows that are disabled will assume the previous row value.
		</p>

        <h3>Steps to reproduce the problem:</h3>
        <ol>
        	<li>In the first row, click on "click me" link. disabled evaluates to true, and it will not show confirmDialog. Below dataTable will appear the message "You clicked: Mojarra". So far so good.</li>
        	<li>On the second row, disabled evaluates to false, so the confirmDialog will appear. Click yes. Below dataTable will appear "You clicked: Myfaces". OK</li>
        	<li>Click on the fifth row (Spring). It will show a confirmDialog, but it should not. Click yes. Below dataTable, will appear <b>"You clicked: Hibernate"</b> Note that the fourth row assume the fifth row place.</li>        
        </ol>
        
        <p:confirmDialog id="globalConfirmDialog" global="true" closeOnEscape="true" >
		<p:commandButton value="Yes"  type="button" styleClass="ui-confirmdialog-yes" icon="ui-icon-check"/>  
       		<p:commandButton value="No"   type="button" styleClass="ui-confirmdialog-no"  icon="ui-icon-close"/>       
	</p:confirmDialog> 
        
        <h:form>
        	
        	<p:dataTable id="dTable" value="#{testConfirmView.rows}" var="rec" >
				 
				 <p:column headerText="Action">
				 	<p:commandLink process="@form" update="@form" value="click me">
				 		<f:setPropertyActionListener value="#{rec}" target="#{testConfirmView.selRow}" />
						<p:confirm header="Confirm" message="You will click #{rec.name}. Proceed?" icon="ui-icon-alert" disabled="#{not rec.showConfirm}" />
					</p:commandLink>
				 </p:column>
				 			 
			     <p:column headerText="Should show confirm dialog?" style="width: 200px;">
			     	<h:outputText value="#{rec.showConfirm}" />
			     </p:column>
				 			     
			     <p:column headerText="ID" style="width: 100px;">
			     	<h:outputText value="#{rec.id}" />
			     </p:column>
			     
			     <p:column headerText="Name" >
			     	<h:outputText value="#{rec.name}" />
			     </p:column>
				 
			</p:dataTable>
				
			<br/>
			<br/>
			
			<h:outputText value="You clicked: #{testConfirmView.selRow.name}" />
			
        </h:form>

    </h:body>
</html>
```

## 6) Sample bean
```java
package org.primefaces.test;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;

import javax.annotation.PostConstruct;
import javax.faces.bean.ManagedBean;
import javax.faces.bean.ViewScoped;

@ManagedBean(name = "testConfirmView")
@ViewScoped
public class TestConfirmView implements Serializable {
     
    private static final long serialVersionUID = 1L;
	
    private Record selRow;
    private List<Record> rows;
    
    @PostConstruct  
    public void init() {
    	rows = new ArrayList<>();        
    	
        rows.add(new Record(1, "Mojarra"));
        rows.add(new Record(2, "Myfaces"));
	rows.add(new Record(3, "Eclipselink"));
	rows.add(new Record(4, "Hibernate"));
	rows.add(new Record(5, "Spring"));        
	rows.add(new Record(6, "Primefaces"));        
    }

	public Record getSelRow() {
		return selRow;
	}

	public void setSelRow(Record selRow) {
		this.selRow = selRow;
	}

	public List<Record> getRows() {
		return rows;
	}

	public void setRows(List<Record> rows) {
		this.rows = rows;
	}
    
}

```

```java
package org.primefaces.test;

public class Record {

	private Integer id;
	private String name;
	
	public Record() {
	}

	public Record(Integer id, String name) {
		super();
		this.id = id;
		this.name = name;
	}
	

	public Integer getId() {
		return id;
	}

	public void setId(Integer id) {
		this.id = id;
	}

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}
	
	public boolean isShowConfirm() {
		return id % 2 == 0;
	}
	
}

```

## Sample image from reproducer:
![bug_reproducer_confirmdialog](https://user-images.githubusercontent.com/7378267/46576281-62889700-c99c-11e8-89ce-eee57d461d30.png)

**Comment by Rapster** [comment]
@lagar84 could you have a try again, looks like it has been fixed

**Comment by melloware** [comment]
Good catch @Rapster it looks like this was fixed in 6.2.12.

**Comment by lagar84** [comment]
@Rapster @melloware  I ran the reproducer using PrimeFaces 6.3-SNAPSHOT - indeed now it is working as expected. it is safe to close this issue now.
